### PR TITLE
Zoom sensitivity ratio tweak, header file modifications

### PR DESCRIPTION
**Pull request by Zanthed**
Solves https://github.com/nullworks/cathook/issues/1427 and https://github.com/nullworks/cathook/issues/958

Signed-off-by: Ashley <ash@trapacid.dev>

### CODE REVIEW

**Review by iraizo — DISMISSED** [review]
LGTM

**Comment by LightPower392** [comment]
I think this should have been done differently. It always resets zoom_sensitivity_ratio to 1, even if the player had manually set it to a different value previously. It should also only change your sensitivity if "Remove scope" and "Remmove zoom/keep rifle" are enabled

**Comment by BenCat07** [comment]
It should probably be a seperate option aswell.

File: src/hooks/visual/OverrideView.cpp [hunk 1]
```diff
@@ -119,6 +120,14 @@ DEFINE_HOOKED_METHOD(OverrideView, void, void *this_, CViewSetup *setup)
...
 
     draw::fov = setup->fov;
+
+    static auto zoom_sensitivity_ratio = g_ICvar->FindVar("zoom_sensitivity_ratio");
+
+    if (zoom_sens_fix)
+        if (zoomed)
+            zoom_sensitivity_ratio->SetValue(4);
+        else
+            zoom_sensitivity_ratio->SetValue(1);
```
Comment (BenCat07) [inline src/hooks/visual/OverrideView.cpp:130]: Does not reset to user defined Value.

**Review by BenCat07 — CHANGES_REQUESTED** [review]

**Comment by relaybots** [comment]
I cannot tell you how long iv waited for this, but holy fuck is it good

**Review by TotallyNotElite — CHANGES_REQUESTED** [review]
Wtf

**Comment by iraizo** [comment]
Getting value before setting it would fix the issue, no?

**Comment by BenCat07** [comment]
No, the code constantly sets sens, even if zoom state did not change at all. Please fix that.

**Comment by iraizo** [comment]
Yeah I've talked to him about that, but didn't find a proper solution since I rather looked into something that only sets it's once Everytime you scope and not based on if the sensitivity changes

**Comment by BenCat07** [comment]
update on zoom state change instead of every tick

**Comment by LightPower392** [comment]
It still has the problems that `zoom_sens_fix` doesn't have actually have to be enabled for zoom_sensitivity_ratio to be changed. This is how I would have written this

    static auto zoom_sensitivity_ratio = g_ICvar->FindVar("zoom_sensitivity_ratio");
    
    if (g_pLocalPlayer->bZoomed)
    {
        // At least one of the three requirements are false, so user zoom_sensitivity_ratio will be reset to what they had it to
        if (!zoom_sens_fix || !no_zoom || !no_scope)
        {
            if (user_sensitivity_ratio_set)
                zoom_sensitivity_ratio_user = zoom_sensitivity_ratio->GetFloat();
            else
            {
                zoom_sensitivity_ratio->SetValue(zoom_sensitivity_ratio_user);
                user_sensitivity_ratio_set = true;
            }
        }
        // All three requirements are true, so change the zoom_sensitivity_ratio to 4
        else
        {
            zoom_sensitivity_ratio->SetValue(4);
                user_sensitivity_ratio_set = false;
        }
    }
And put this above `namespace hooked_methods`:
    `bool user_sensitivity_ratio_set = true;`
    `float zoom_sensitivity_ratio_user;`

**Comment by Trapacid** [comment]
Thank you, @LightPower392 lmao.

Is that better?

**Comment by LightPower392** [comment]
That looks good except that you should probably update the tooltip for `Remove zoom/keep rifle` to reflect the change

**Comment by advance24th** [comment]
im trying to compile these edits to cathooks 
and im getting an output at the end like 

```
make[3]: *** [CMakeFiles/cathook.dir/build.make:1805: CMakeFiles/cathook.dir/src/hooks/CreateMove.cpp.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/Makefile2:646: CMakeFiles/cathook.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:653: CMakeFiles/cathook.dir/rule] Error 2
make: *** [Makefile:182: cathook] Error 2
```

am i missing somthing?

**Comment by Trapacid** [comment]
Edited the tooltip.

I did not modify CreateMove.cpp in any way. @advance24th

**Comment by advance24th** [comment]
weird..

**Comment by advance24th** [comment]
the error is

```
/home/notgonnagivemyname/cathook/src/hooks/CreateMove.cpp:307:25: error: ‘fakelag_midair’ was not declared in this scope
  307 |                     if (fakelag_midair && CE_INT(LOCAL_E, netvar.iFlags) & FL_ONGROUND)
      |                         ^~~~~~~~~~~~~~
```

**Comment by advance24th** [comment]
everything till that code, was fine so i know its just this line of code

**Comment by Trapacid** [comment]
Please create a separate issue for your compilation error.

**Comment by advance24th** [comment]
please just this one question

is this [error fix(#1428)](https://github.com/nullworks/cathook/pull/1428) for the stable version of cathook?
or master(bleeding edge)?

**Comment by Trapacid** [comment]
@advance24th Master, but do not use my branches/repo as they are not up to date with upstream.

If you wish to test this patch:
Clone the upstream repo in a test folder. Run `./config` and switch it to developer mode. Build and install cathook as normal with `./install-all`.
Then download the patch: `wget https://github.com/nullworks/cathook/commit/ea96ade96ae9a054f29bba15895686cd146923e7.patch`
Apply it with: `git apply ea96ade96ae9a054f29bba15895686cd146923e7.patch`
Run `make data && make`
Inject: `sudo ./attach`

**Comment by advance24th** [comment]
Thanks ill try that, but for what i did before you commented was in this Issue ^^^^

**Comment by iraizo** [comment]
Closes https://github.com/nullworks/cathook/issues/1427

**Comment by advance24th** [comment]
> Closes #1427

i got it fixed anyways, goodbye and send this in regard to me

### Im dumb as fuck

**Comment by advance24th** [comment]
wamo you delete my comment, about me saying im dumb

**Comment by relaybots** [comment]
I hate yiu people who dare remove hate speech about me :(

**Comment by BenCat07** [comment]
Currently this:
- Breaks if you enable both rvars and zoom in before having zoomed in "normally" once (it will set the zoom sens to 0 once you use normal zooming)
- Constantly updates the sens while zoomed, even if the state didn't change

I fixed both those issues for you, so the pull request should be ready.

**Review by BenCat07 — APPROVED** [review]
LGTM

**Pull request merged**